### PR TITLE
Data graph resolution

### PR DESCRIPTION
**Pull request by upatricck**
Added start time and end time for the data analyssis graph,
Also added a resolution to specify,  hourly , daily, weekly or monthly

### CODE REVIEW

**Comment by jlark** [comment]
Does this also have the CSV data fixes applied?

**Comment by upatricck** [comment]
Yeah it does

File: seshdash/data/db/influx.py [hunk 1]
```diff
@@ -315,6 +313,35 @@ def get_measurement_range(self, measurement_name, start, end, site=None, databas
   
         return results
 
+    def get_measurement_range_bucket(self, measurement_name, start, end, group_by, site=None, database=None):
+        """
+        Returns the measurement points for a given range grouped by a given time
+
+        @param measurement_name - name of the measurement
+        @param start - The date to start from (type python datetime)
+        @param end - The time to stop at ( type python datetime)
```
Comment (jlark) [inline seshdash/data/db/influx.py]: params for site and database are missing

database should be the database defined in configs by default.
something  like.
```
if not database:
   database = settings.INFLUX_DATABASE
```

File: seshdash/static/sesh-ui/js/graphs.js [hunk 2]
```diff
@@ -1,35 +1,55 @@
...
+          'start-time': start_time,
+          'end-time': end_time,
+          'active_site_id': active_site_id,
+          'resolution': resolution
+    }
+
+
+    $.get("/graphs", data_to_send, function(data){
+
+        console.log(data);
```
Comment (jlark) [inline seshdash/static/sesh-ui/js/graphs.js]: let's remove the console.log

File: seshdash/views.py [hunk 3]
```diff
@@ -796,62 +797,81 @@ def historical_data(request):
...
+    results = []
 
-    # if ajax request
-    if request.method == 'POST':
+    # Getting values from Post request
+    time = request.GET.get('time', '') # This is the time range it has to be: 24h, 7d or 30d
+    choices = request.GET.getlist('choice[]')
+    active_id = request.GET.get('active_site_id', None)
+    start_time = request.GET.get('start_time', None)
+    end_time = request.GET.get('end_time', None)
```
Comment (jlark) [inline seshdash/views.py]: default for end time needs to be now. We shouldn't be able to request an endtime in the future as well

File: seshdash/views.py [hunk 4]
```diff
@@ -796,62 +797,81 @@ def historical_data(request):
...
+    resolution = request.GET.get('resolution', '1h')
+    current_site = Sesh_Site.objects.filter(id=active_id).first()
+    
+
+    print "Got values %s %s %s of site %s " % (time, choices, active_id, current_site)
```
Comment (jlark) [inline seshdash/views.py]: print remove please: )

File: seshdash/views.py [hunk 5]
```diff
@@ -796,62 +797,81 @@ def historical_data(request):
...
+    if not current_site:
+        return HttpResponseBadRequest("Invalid site id, No site was found for the given site id")
```
Comment (jlark) [inline seshdash/views.py]: we  should also check the user has permission to view the site.
Else someone can send a GET request for a site the don't have permission to view and still get data.

File: seshdash/views.py [hunk 6]
```diff
@@ -816,7 +816,8 @@ def graphs(request):
     current_site = Sesh_Site.objects.filter(id=active_id).first()
     
 
-    if (not current_site) or site.organisation != user.organisation:
+    if (not current_site) or current_site.organisation != request.user.organisation:
+        print "The site is: %s" % current_site
```
Comment (jlark) [inline seshdash/views.py]: print ;0

**Comment by jlark** [comment]
LGTM

**Pull request merged**